xresolver/xresolver.go

```diff
@@ -0,0 +1,113 @@
+package xresolver
+
+import (
+	"context"
+	"errors"
+	"net"
+	"strconv"
+	"sync"
+)
+
+// Note to self: Dial is not being set for net.Resolver because that is the Dial to the DNS server.
+
+var DefaultDialer = &net.Dialer{}
+
+type resolver struct {
+	resolvers map[Lookup]struct{}
```

Comment:
If you are trying to do a set, use `map[Lookup]bool`.  That's the standard `golang` idiom for a set of objects.  You can then rely on zero-values: `resolvers[...]` will return `false` if no such lookup is in the map, `true` if it is.

See https://golang.org/doc/effective_go.html#maps